Implementation/model/beer

TheNoobProgrammeur opened the pull request:
Mise en place des models

Commit:
:sparkles: implement User model

Commit:
:heavy_minus_sign: remouve jupiter lib

Commit:
:white_check_mark: Mouve test

Commit:
:white_check_mark: fix test

Commit:
:construction: Implementation lecture/ecriture de la class user en Json

Commit:
:construction:  changerment de type pour BeerList

Commit:
✨ implement Beer model

Commit:
🚧  ajout des class

Commit:
🚧  Update class

Commit:
Merge branch 'implementation/model/User' into implementation/model/Beer

Commit:
🚧  Update class

Commit:
🚧  Update constructeur  class

Commit:
🚧  Update  Bitmap

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/TypeBeer.kt:
```diff
@@ -0,0 +1,14 @@
+    IPA,
+    AMBREE,
+    BLONDE,
+    FONCEE_
```
underscore useless

acroquelois replied:
FONCEE(_)

TheNoobProgrammeur replied:
Jolie faute de frappe

TheNoobProgrammeur replied:
fix

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/Beer.kt:
```diff
@@ -0,0 +1,46 @@
+        this.type = typeBeer
+    }
+
+    constructor(beer: Beer) : this() {
```
A quoi sert ce constructeur ?

TheNoobProgrammeur replied:
Lorsqu'on fait evoluer une `Beer` en `BeerDescription`  on passe la Beer en param pour rec les infos. ce constructeur est utiliser comme methode super

acroquelois replied:
oui je vois, je ferme la review pour ce fichier alors

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/Beer.kt:
```diff
@@ -0,0 +1,46 @@
+        this.average = averger
+    }
+
+    constructor(id:String, name:String, degree:Double, brand: Brand, typeBeer: TypeBeer) : this() {
```
On lieu de faire plusieurs constructeur on peut pas plutôt en faire un seul avec des valeurs par défaut qd la valeur n'est pas renseigné en param  ? (Pour les valeurs non required) Après c'est du no sql donc justement c'est peut être mieux de ne pas définir les attributs qd ils sont null comme tu l'as fait, en faisant plusieurs constructeurs

TheNoobProgrammeur replied:
ha oui en effet

acroquelois replied:
Tu m'as pas répondu ducoup xD

TheNoobProgrammeur replied:
Nop, mon assiette ma appelé 😁😁😁

TheNoobProgrammeur replied:
Un seul constructeur serait plus intéressant. Car si on stock que les champs remplie alors on mettra que les champs qu'on récupère.

TheNoobProgrammeur replied:
Commentaire pris en compte

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/BeerDetail.kt:
```diff
@@ -0,0 +1,11 @@
+    lateinit var description : String
+    lateinit var brewery : Brewery
+
+    constructor(beer: Beer, description : String, brewery: Brewery):super(beer){
```
Ici je me demande toujours, si tu veux créer un BeerDetail à partir de zero t'es obligé de d'abord créer une Beer ? Il faudrait pas qu'on puisse directement créer une BeerDetail avec tt ses argument ? Ca pourrait être plus pratique au moment ou on recuperera le get/{id} de la bdd, on pourra direct balancer le result de la requête dans le constructeur BeerDetail. Tu en penses quoi ?

TheNoobProgrammeur replied:
On peut créer un constructeur de ce type.
Si on reprend ta remarque plus haut ça nous fait 2 constructeur

acroquelois replied:
Oui pour moi ici les deux constructeurs nous aiderons plus tard au moment de faire les requêtes à la bdd

TheNoobProgrammeur replied:
Commentaire pris en compte

Commit:
🚧  Simplification des constructeur

Commit:
🚧  Simplification des constructeur fix test

Commit:
🚧  Passage de class en Data class

Commit:
🚧  Del constructeur

Commit:
🚧  Add Comment

Commit:
🚧  Update Test

Commit:
🚧  Update class

acroquelois committed:
:construction: modif class Beer

acroquelois committed:
Merge remote-tracking branch 'origin/implementation/model/Beer' into implementation/model/Beer

acroquelois committed:
:twisted_rightwards_arrows: resolve conflict

Commit:
🚧 Fix class

- Prise en compte du bug sur le vote : les user on une liste des id des biere noter 
- Update de l'enum TypeBeer 
- Fix des TU
- Travaille sur les constructeur

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/User.kt:
```diff
@@ -0,0 +1,17 @@
+    var beerFavorite: ArrayList<Beer> = ArrayList<Beer>(),
+    var beerNote: ArrayList<String> = ArrayList<String>()
+) {
+    lateinit var image:Bitmap
```
Pour moi image c'est à voir dans un second temps, on sait pas encore comment les médias seront géré

TheNoobProgrammeur replied:
dans le storege de firebase. Donc c'est plutot un ID ==> String

acroquelois replied:
Je serai d'avis de ne pas penser à ça tout de suite

TheNoobProgrammeur replied:
Oki, a valider avec @BBR2394

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/Comment.kt:
```diff
@@ -0,0 +1,6 @@
+
+import java.util.*
+
+data class Comment(val id:String, val idUser:String, val idBeer:String, val date:Date, val message:String) {
```
Je pense qu'on aura un username plutot qu'un idUser ici

TheNoobProgrammeur replied:
Oui je suis d'accord, simplification des reqs, j'aime ca ><

acroquelois replied:
Je me met à penser noSQL xDD

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/User.kt:
```diff
@@ -0,0 +1,17 @@
+    var mail: String,
+    var id: String,
+    var beerFavorite: ArrayList<Beer> = ArrayList<Beer>(),
+    var beerNote: ArrayList<String> = ArrayList<String>()
```
Class note à crée avec un idBeer et un float sinon comment associer la note à la bière ?

TheNoobProgrammeur replied:
Il faut pas associer une note a la bière, il faut simplement que le user se souvienne des bières qu’il a notées. Sauf si tu veux aussi qu’il se souvienne de la note pour un éventuel affichage.

acroquelois replied:
En tant qu'utilisateur qd tu cliques sur une bière tu aimes savoir cb tu lui as mis :smile:

TheNoobProgrammeur replied:
Oki, c'est partie pour une `data class notes`

TheNoobProgrammeur reviewed (commented):
réponse a @acroquelois

acroquelois commented on app/src/main/java/berthier/antoine/birudroid/model/Beer.kt:
```diff
@@ -0,0 +1,16 @@
+import java.lang.reflect.Type
+
+open class Beer(
+        var id: String = "ID_NOT_DEF",
```
Ah oui aussi, pq ces valeurs par defaut ?

TheNoobProgrammeur replied:
pb de constructeur. notament pour le super de la class `beerdetail`

Commit:
🚧 Implementation data class Note

Commit:
✅ fix test

Commit:
:fire: Del image user
